pkg/infradb/utils.go

```diff
@@ -0,0 +1,294 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (c) 2023-2024 Intel Corporation, or its subsidiaries.
+// Copyright (c) 2024 Ericsson AB
+
+// Package infradb exposes the interface for the manipulation of the api objects
+package infradb
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/opiproject/opi-evpn-bridge/pkg/infradb/subscriberframework/actionbus"
+	"github.com/opiproject/opi-evpn-bridge/pkg/infradb/subscriberframework/eventbus"
+	"github.com/opiproject/opi-evpn-bridge/pkg/infradb/taskmanager"
+)
+
+func startReplayProcedure(componentName string) {
+	globalLock.Lock()
+
+	var deferErr error
+	var preSubscriber *actionbus.Subscriber
+
+	defer func() {
+		if deferErr != nil {
+			log.Println("startReplayProcedure(): The replay procedure has failed")
+			log.Println("startReplayProcedure(): unblocking the TaskManager to continue")
+			taskmanager.TaskMan.ReplayFinished()
+		}
+	}()
+
+	preSubscribers := actionbus.ABus.GetSubscribers("preReplay")
+
+	for _, preSub := range preSubscribers {
+		if preSub.Name == componentName {
+			preSubscriber = preSub
+			break
+		}
+	}
+
+	if preSubscriber == nil {
+		deferErr = fmt.Errorf("no pre-replay subscriber for %s", componentName)
+		log.Printf("startReplayProcedure(): Error %+v\n", deferErr)
+		return
```

Comment:
global lock  won't be released if we exit here, right?

Reply:
I think you are right on this. I will investigate

Reply:
I think you are right on this.  I will investigate

Reply:
Done

Reply:
looks convoluted. We unlock in 2 different places. In one if there is an error, in another if there is no error.
I would move the code where we need to keep lock in a separate function where just `defer mu.Unlock()` is used

Reply:
I have changed it. I think now is better